examples/bin/start-druid

```diff
@@ -0,0 +1,548 @@
+#!/usr/bin/env python3
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import sys
+import os
+import psutil
+import pathlib
+import multiprocessing
+import argparse
+
+QUICKSTART_ROOT_CONFIG_PATH = "conf/druid/single-server/quickstart"
```

Comment:
If I understand this correctly these configs will be used as a base and then they will be "extended" when the server starts using the info from the `--memory` config. If so I think we should not call these configs "quickstart". To me "quickstart" means the shitty cluster you spin up on your laptop just to check stuff out (vs the "real" cluster you will deploy later). If my understanding is correct I would call these "base" configs or something similar because you might actually go into production with your start script being `bin/start-druid -m 64g` or something light that.

Also this is not really single server right? You might run `bin/start-druid -s broker,router` on once server and then `bin/start-druid -s historical,middleManager` and `bin/start-druid -s coordinator-overlord` on other servers to make a multi-server cluster.

I would have the configs live in a new location like `conf/druid/base` or `conf/druid/dynamic`

Reply:
Makes sense. We might eventually get rid of `single-server` and `cluster` directories anyway, since the new script would become the single entry point for everything.

Reply:
> If I understand this correctly these configs will be used as a base and then they will be "extended" when the server starts using the info from the --memory config.

yes, that is correct. 

> If my understanding is correct I would call these "base" configs or something similar because you might actually go into production with your start script being bin/start-druid -m 64g or something light that.

yes, I think `base` would be a better name. 

> Also this is not really single server right? You might run bin/start-druid -s broker,router on once server and then bin/start-druid -s historical,middleManager and bin/start-druid -s coordinator-overlord on other servers to make a multi-server cluster.

yes, that is correct. 

In the very first commit I had a new directory under `conf/druid` with a similar name, it was moved under `single-server` since we were targeting compatibility with single-server launch scripts. But I think it will be better, if these configs live under `conf/druid/base`.

Reply:
Makes sense to me too, in which case we can alter my suggestions for the `single-server.md` page; instead of doing `conf/druid/single-server/auto` it would be `conf/druid/base`.

Or maybe `start-druid-base`, so it can be on-its-face obvious that it's connected to the `bin/start-druid` script?

I am ok with a variety of options here, so as far as I am concerned, feel free to do what you think makes the most sense given what we have discussed.

Reply:
I am calling the script `start-druid-auto` and the config directory is `conf/druid/auto`. 

auto (or dynamic) was conveying the capabilities of the script better than `base`. 
Since `auto` was short I picked it over `dynamic`.